### ISSUE TITLE
Rewrite the VR plugin

### ISSUE THREAD

**Issue by edreamleo** [leo-editor/leo-editor#3892]
See PR #3893 for all details.

The rewrite greatly simplifies the code and adds important features.

**Comment by tbpassin** [comment]
I don't this the proposal as it stands gets at the user/UI needs yet.  I think the process of showing a rendering pane needs to be simple for the user, easily changeable, and though defaults may be good, defaults can end up being an annoying constraint.  In addition, a single setting will not always be appropriate for all nodes and documents.  Most people will want one default for one kind of node and another for a different kind of node.  Also the desired default may change during the life cycle of a node.  For example, when I am authoring a notebook I want to have an editing mode be the normal, but once I am mostly done authoring I want a rendering mode to be the default. Either way, it needs to be fast and easy to change the mode back and forth.

There is also a real possibility that having core code do automatic switching based on language types may end up conflicting with VR/VR3 ideas of what the node type should be - that's the kind of thing that happens over time as code changes.

Let's think about how some other code editors, let's say VSCode, work.  Each document gets a tab of its own, but rendering views also get their own tabs.  Viewing a rendered view takes only clicking on the right tab, or at most creating a new tab for the rendering view.  The actual display normally takes place in the same part of the window for all the tabs.  VSC remembers (IIRC) which tab was active next time it opens.

VSC has a Markdown rendering extension, and an RsT one (actually, several of each, I think).  They don't connect to LeoJS so they can't be used to render Leo nodes.  But they give the idea.

IMHO, if we're not going to give the Leo body frame multiple tabs like that (perhaps it would be too hard), then there could be a command or button to show the rendering pane on top of the editing view, and a simple way to switch between them.  Other commands/buttons could open a rendering view in a new frame or even a free-floating window.

VR3 at present has a command to toggle (or create) a VR3 view in a separate frame and another to toggle (or create) a VR3 view in a tab in the log frame.  We know it takes just two lines of code in a button to open a free-floating VR3 window.  It seems natural to think about adding a similar command to add or toggle a VR3 view on top of the editing view.  I don't think any other commands or settings would be needed.  If we decide we want to persist the state of the rendering view, then save it when an outline or Leo closes like we do, say, the selected position in the tree.

Currently VR3 can be set to be active or inactive when editing is done on the underlying node.  We shouldn't lose that ability because sometimes it's just too slow or annoying to update the rendering during editing.

**Comment by tbpassin** [comment]
It's already possible to get the splitter to put the body editor into the log frame, and VR3/VR can be put there too.  That's not ideal because any write to the log panel switches away from the VRx/body pane which is not really what one wants.

To implement this concept, all we need, I think, is to put a frame with the Log frame machinery where the body editor now goes, but to build it without a Log pane.  One strength of this machinery is that any plugin with a display widget can be displayed in the frame or removed with minimal code.  It's nearly trivial to connect this code to a minibuffer command so that it can be activated by button or user menu item (I do both).

Here's the code I use to add/toggle my bookmark manager to a tab in the log frame.  The Manager's code itself knows nothing about any of this:
```
"""Create or toggle Bookmark tab in Log pane."""

import bookmark_search_panel
top_widget = bookmark_search_panel.PanelWidget
log = c.frame.log

TABNAME = 'Bookmarks'

def toggle_app_tab(log, tabname):
    """Create or remove our app's tab.
    
    ARGUMENTS
    log -- the log panel object for this outline.
    tabname -- a string to use as the display name of our tab.
    top_widget -- an instance of top level widget of our app.
    """
    VISIBLE = f'{tabname}-visible'
    LOADED = f'{tabname}-loaded'
    WIDGET_NAME = f'{tabname}-widget'

    # If our tab is visible, remove it
    if log.contentsDict.get(VISIBLE, False):
        log.deleteTab(tabname)
        log.contentsDict[VISIBLE] = False
        font = c.config.getColor('font-family')
        editor = c.frame.body.wrapper.widget
        editor.setStyleSheet(f'QTextBrowser {{font-family: {font};}}')
    else:
        # Show our tab, reusing our widget if already loaded
        if log.contentsDict.get(LOADED, False):
            log.createTab(tabname,
                          widget = log.contentsDict[WIDGET_NAME],
                          createText = False)
            log.contentsDict[VISIBLE] = True
            log.selectTab(tabname)
        else:
            # Create our tab for the first time
            w = top_widget(g, c)
            log.createTab(tabname, widget = w, createText = False)
            log.selectTab(tabname)
            log.contentsDict[LOADED] = True
            log.contentsDict[VISIBLE] = True
            log.contentsDict[WIDGET_NAME] = w

        editor = c.frame.body.wrapper.widget
        editor.setStyleSheet("""\
QTextBrowser {
    font-family: Corbel, Verdana, "Open Sans", Sans-serif; 
    font-size: 12pt;
}""")

toggle_app_tab(log, TABNAME)
```

**Comment by edreamleo** [comment]
Closed via PR #3893. Later PRs are likely.
